### PR TITLE
fix: encode full filter parameter once

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Moved the encode method up to the last point when the filter string is returned.

### CODE REVIEW

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
Looks good for the filter, but we also want to encode the path and other query parameters.

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 ## Fixed Issues
 
 - [PROXY] Consider the web proxy also in the service call to BTP services like XSUAA and destination service.
+- [odata] Fix encoding of filter strings.
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply CHANGELOG.md:30]: I thought about this but it seemed not to be necessary. So the [url](https://github.com/SAP/cloud-sdk-js/blob/74815776f7efa95fae93566fca166de2cacc64cb/packages/core/src/odata-common/request/odata-request.ts#L60) contains only the [query](https://github.com/SAP/cloud-sdk-js/blob/74815776f7efa95fae93566fca166de2cacc64cb/packages/core/src/odata-common/request/odata-request.ts#L145) part which needs to be encoded. I went than to the get all [implementation](https://github.com/SAP/cloud-sdk-js/blob/74815776f7efa95fae93566fca166de2cacc64cb/packages/core/src/odata-common/request/odata-get-all-request-config.ts#L41-L59) which contained the:
- select
- expand
- orderBy
- filter
as parts. I think only the filter could contain parts which need encoding. Our naming convention in the generator and restrictions in TS should make encoding for the other parts not necessary.

[the remark this answers is not in the document]
Comment (jjtang1985) [reply CHANGELOG.md:30]: ### query parameters
- question: how about properties containing umlaut characters, which might be used by `select`/`expand`/`orderBy`?
- we also have custom query parameters, which might be `?objectName=' space space '`.
### path
- service path/custom service path might need encoding

[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply CHANGELOG.md:30]: For the custom path and parameter I would say we should keep them as they are. Regarding the special characters in select, expand this is true in principal.

**Review by jjtang1985 — APPROVED** [review]
As discussed, we cannot encode the whole URL, as the custom part should not be encoded. Splitting the encoding job is also error prone, so we leave it for now and wait until further feedbacks.

**Comment by jjtang1985** [comment]
related to https://github.com/SAP/cloud-sdk-js/issues/1226

**Pull request merged**